File: python/bubble_sort.py
# sequence = [4, 3, 5, 0, 1]
# swaps = 0

def bubblesort(sequence):
    swapped = False
    # Looping from size of array from last index[-1] to index [0]
    for k in range(len(sequence)-1, 0, -1):
        for j in range(k):
            if sequence[j] > sequence[j + 1]:
                swapped = True
                # swapping data if the element is less than next element in the array
                sequence[j], sequence[j + 1] = sequence[j + 1], sequence[j]
                print(sequence)	
            if not swapped:
                print(sequence)
                # exiting the function if we didn't make a single swap
                # meaning that the array is already sorted.
                return

sequence = [39, 12, 18, 85, 72, 10, 2, 18]

# print("Unsorted list is,")
# print(sequence)
print(bubblesort(sequence))
# print("Sorted Array is, ")
# print(sequence)


# print(f"Final result: {result}")
# print(f"Swaps: {swaps}")

File: js/bubbleSort.js
var sequence = [4, 3, 5, 0, 1]
var swaps = 0

function bubbleSort(sequence){	
    let swaps = 0;	
    let iterations = 0;	
    for(let i = 0; i < sequence.length; i++){	
      for(let j = 0; j < ((sequence.length - i) - 1); j++){	
        iterations++	
        if(sequence[j] > sequence[j + 1]){	
          let temp = sequence[j];	
          sequence[j] = sequence[j + 1];	
          sequence[j + 1] = temp;	
          swaps++;        	
        }	
      }	
    }	
    return [sequence, swaps, iterations]	
  }

console.log(bubbleSort(sequence))
// console.log("Final result: " + result)
// console.log("Swaps: " + swaps)


// function bubbleSort(sequence){	
//     let swaps = 0;	
//     let iterations = 0;	
//     for(let i = 0; i < sequence.length; i++){	
//       let swapped = false;	
//       for(let j = 0; j < ((sequence.length - i) - 1); j++){	
//         iterations++	
//         if(sequence[j] > sequence[j + 1]){	
//           let temp = sequence[j];	
//           sequence[j] = sequence[j + 1];	
//           sequence[j + 1] = temp;	
//           swapped = true;	
//           swaps++;        	
//         }	
//         if (!swapped){	
//           break;	
//         }	
//       }	
//     }	
//     return [sequence, swaps, iterations]	
//   }